amp-update-cache/cache-refresh.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const fetch = require('node-fetch');
+const jsrsasign = require('jsrsasign');
+const {URL} = require('url');
+const mime = require('mime-types');
+const punycode = require('punycode');
+
+const CACHES_JSON_URL = 'https://cdn.ampproject.org/caches.json';
+
+class CacheRefresh {
+  constructor(privateKey) {
+    this._privateKey = privateKey;
+    this._sig = new jsrsasign.Signature({'alg': 'SHA256withRSA'});
+    this._sig.init(this._privateKey);    
+  }
+
+  createCacheUpdateUrls(originUrl) {
+    return this._getCaches()
+      .then(caches => {
+        return caches.map(cache => {
+          const cacheUrl = this.createCacheUrl(originUrl, cache);
+          const refreshUrl = this.createRefreshUrl(cacheUrl);
+          return {id: cache.id, name: cache.name, refreshUrl: refreshUrl}
+        });
+      });
+  }
+
+  /**
+   * Translates an url from the origin to the AMP Cache URL format, as documented here:
+   *  https://developers.google.com/amp/cache/overview
+   *
+   * @param {String} originUrl the URL to be transformed.
+   * @param {Object} cache the cache Object containing the suffix configuration.
+   * @return {String} the transformed URL.
+   */
+  createCacheUrl(originUrl, cache) {
```

Comment:
You could also use the official converter from: `https://cdn.ampproject.org/viewer/google/amp_url.js`, usage example [here](https://github.com/ampproject/amp-by-example/blob/master/api/scripts/amp_url_converter.js#L42).